airflow/models.py

```diff
@@ -1288,19 +1288,19 @@ def run(
             verbose=True)
 
         if not runnable and not mark_success:
-            if self.state != State.QUEUED:
-                # If a task's dependencies are met but it can't be run yet then queue it
-                # instead
-                self.state = State.QUEUED
-                msg = "Queuing attempt {attempt} of {total}".format(
-                    attempt=self.try_number % (task.retries + 1) + 1,
-                    total=task.retries + 1)
-                logging.info(hr + msg + hr)
-
-                self.queued_dttm = datetime.now()
-                msg = "Queuing into pool {}".format(self.pool)
-                logging.info(msg)
-                session.merge(self)
+            # we might have hit concurrency limits, which means we probably
+            # have been running prematurely.
+            self.state = State.UP_FOR_RETRY
```

Comment:
Well maybe. It is actually true that the task instance was running and as we are hitting a RUN_DEPS limit (90%) it also needs to back off for a bit of time. This also allows for proper logging. Otherwise we need to go to NONE or SCHEDULED both of which should imho not be set by a task instance as they are part of the scheduler, thus this would require a new state. 

I think this is the cleanest solution.